react/src/component/login/login.tsx

```diff
@@ -0,0 +1,217 @@
+            return <LoginEmailConfirm needCompleteRegistration={needCompleteRegistration} token={token}/>
+        default:
+            throw new Error(`could not handle login method=${method}`)
+    }
+}
+
+
+
+export const LoginNeedCompleteRegistration = (props) => {
+    let appName = useContext(CAppName);
+    let navigate = useNavigate();
+    let onCompleteForm = (data: {[key: string]: any}) => {
+        fetch(`/auth-complete-registration`, {
+            method: "POST",
+            body: JSON.stringify({ ...data, simulationType: appName }),
+            headers: {
+                "Content-Type": "application/json"
+            }
+        }).then(() => navigate(`/react/${appName}`))
+    }
+    return (
+        <LoginExtraInfoForm onComplete={(data) => onCompleteForm(data)}/>
+    )
+}
+
+export const LoginExtraInfoForm = (props: { onComplete: ({displayName}) => void }) => {
+    let { onComplete } = props;
+    let [displayName, updateDisplayName] = useState<string>("");
+    return (
+        <>
+            <p>Please enter your full name to complete your Sirepo registration.</p>
+            <Form.Group as={Row} className="mb-3" controlId="formPlaintextEmail">
+                <Form.Label column sm="3">
+                    Your full name
+                </Form.Label>
+                <Col sm="9">
+                    <Form.Control value={displayName} onChange={(e) => updateDisplayName(e.target.value)}/>
+                </Col>
+            </Form.Group>
+            <Button variant="primary" onClick={() => onComplete({ displayName })}>Continue</Button>
+        </>
+    )
+}
+
+export const LoginRoot = (props) => {
+    let loginStatus = useContext(CLoginStatus);
+    let appName = useContext(CAppName);
+
+    let getLoginComponent = (method: string): JSX.Element => {
+        switch(method) {
+            case "email":
+                return <LoginWithEmail/>
+            case "guest":
+                return <Navigate to={new AppWrapper(appName).getAppRootLink()}/>
+            default:
+                throw new Error(`could not handle login method=${method}`)
+        }
+    }
+
+    if(loginStatus.needCompleteRegistration) {
+        return (
+            <Container className="sm-12 lg-6">
+                <LoginNeedCompleteRegistration/>
+            </Container>
+            
+        )
+    }
+
+    if(loginStatus.isLoggedIn && !loginStatus.needCompleteRegistration) {
+        return (
+            <Navigate to={new AppWrapper(appName).getAppRootLink()}></Navigate>
+        )
+    }
+
+    return (
+        <Container className="sm-12 lg-6">
+            {getLoginComponent(loginStatus.visibleMethod)}
+        </Container>
+    )
+}
+
+export function LogoutRoot(props) {
+    let navigate = useNavigate();
+    let appName = useContext(CAppName);
+
+    useEffect(() => {
+        fetch(`/auth-logout/${appName}`).then(() => navigate(new AppWrapper(appName).getAppRootLink()));
+    })
+    
+
+    return <>Singing out...</>
```

Comment:
One of the big reasons for having a string table is that it can be proofread and spell checked easily.